src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -146,40 +146,6 @@ public override IReadOnlyList<Location> AdditionalLocations
                 get { return _properties; }
             }
 
-            public override bool Equals(Diagnostic obj)
-            {
-                var other = obj as SimpleDiagnostic;
-                if (other == null)
-                {
-                    return false;
-                }
-
-                if (AnalyzerExecutor.IsAnalyzerExceptionDiagnostic(this))
```

Comment:
Analyzer exception diagnostics are special in the sense that they need to be considered equivalent even if their description field (which holds the exception stack trace) differ. This special check avoids any analyzer hosts to require doing this de-duping - they can just maintain a hashset of exception diagnostics and get unique diagnostics. If we remove this, you need to replicate this check at all analyzer host exception handlers (IDE diagnostic service, command line compiler). Additionally, this is a functionality break for any existing 3rd party analyzer hosts or diagnostic consumers - at the minimum we need to expose a public equivalence comparer for Diagnostics that they can use to get the original functionality.